mundane-dsv/src/main/scala/com/ambiata/mundane/dsv/DelimitedValuesParser.scala

```diff
@@ -0,0 +1,68 @@
+package com.ambiata.mundane
+package dsv
+
+import scala.collection.JavaConversions._
+
+import org.apache.commons.csv.{CSVParser, CSVFormat}
+
+import scalaz._, Scalaz._
+
+/**
+ * Create a parser for delimited values based on
+ * different implementations and a choice of
+ *
+ *  - delimiter
+ *  - character used for quoting fields
+ *  - character used to escaped control characters (delimiter, quoteCharacter)
+ *
+ */
+object DelimitedValuesParser {
+
+  def createParser(delimiter: Char, quoteCharacter: Option[Char], escapeCharacter: Option[Char], implementation: DelimitedValuesParserImplementation): DelimitedValuesParser =
+    implementation match {
+      case CommonsCsv => CommonsDelimitedValuesParser(delimiter, quoteCharacter, escapeCharacter)
+    }
+
+  /** use the CommonsCsv implementation as the default implementation */
+  def delimited(delimiter: Char): DelimitedValuesParser =
```

Comment:
This is just thinking out loud, and I know it goes against conventional wisdom, but I would be very tempted to make these not the same type, just make it so they have the same API, I don't know that there is a lot of value in being able to interchange them - and I really _don't_ want to see an end consumer flipping between them - what we have is different use cases demanding different implementations / behaviour - not the desire for a different or configurable behaviour that can cause uncertainty in results.